src/main/java/net/coderbot/iris/shaderpack/IdMap.java

```diff
@@ -59,20 +60,47 @@
 		// TODO: Properly override block render layers
 	}
 
+	/**
+	 * Loads properties from a properties file in a shaderpack path
+	 */
 	private static Optional<Properties> loadProperties(Path shaderPath, String name) {
-		Properties properties = new Properties();
+		if (shaderPath == null) return Optional.empty();
+
+		String fileContents = readProperties(shaderPath, name);
+		if (fileContents == null) return Optional.empty();
 
+		List<String> lines = parseProperties(name, fileContents);
+
+		StringBuilder processed = new StringBuilder();
+		for (String line : lines) {
+			processed.append(line);
+			processed.append('\n');
+		}
```

Comment:
Would probably be best to move this to PropertiesPreprocessor and have parseProperties return a string.